File: Visual Basic .NET/DataVirtualization/Converters/BooleanInverterConverter.vb
' 
' * Xceed DataGrid for WPF - SAMPLE CODE - Data Virtualization Sample Application 
' * Copyright (c) 2007-2024 Xceed Software Inc. 
' * 
' * [BooleanInverterConverter.vb] 
' * 
' * This class implements a IValueConverter which inverts a boolean value. 
' * 
' * This file is part of the Xceed DataGrid for WPF product. The use 
' * and distribution of this Sample Code is subject to the terms 
' * and conditions referring to "Sample Code" that are specified in 
' * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product. 
' * 
' 

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Data
Imports System.Windows

Namespace Xceed.Wpf.DataGrid.Samples.DataVirtualization
  Public Class BooleanInverterConverter
    Implements IValueConverter
#Region "IValueConverter Members"

    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
      If (value Is Nothing) OrElse (Not (TypeOf value Is Boolean)) Then
        Return False
      End If

      Dim inverseValue As Boolean = Not CBool(value)

      Return inverseValue
    End Function

    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
      Throw New NotImplementedException()
    End Function
#End Region

  End Class

End Namespace

File: CSharp/DataVirtualization/Converters/BooleanInverterConverter.cs
/*
 * Xceed DataGrid for WPF - SAMPLE CODE - Data Virtualization Sample Application
 * Copyright (c) 2007-2024 Xceed Software Inc.
 * 
 * [BooleanInverterConverter.cs]
 *  
 * This class implements a IValueConverter which inverts a boolean value.
 * 
 * This file is part of the Xceed DataGrid for WPF product. The use 
 * and distribution of this Sample Code is subject to the terms 
 * and conditions referring to "Sample Code" that are specified in 
 * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 *
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Data;
using System.Windows;

namespace Xceed.Wpf.DataGrid.Samples.DataVirtualization
{
  public class BooleanInverterConverter : IValueConverter
  {
    #region IValueConverter Members

    public object Convert( object value, Type targetType, object parameter, System.Globalization.CultureInfo culture )
    {
      if( ( value == null ) || ( !( value is bool ) ) )
        return false;

      bool inverseValue = !( bool )value;

      return inverseValue;
    }

    public object ConvertBack( object value, Type targetType, object parameter, System.Globalization.CultureInfo culture )
    {
      throw new NotImplementedException();
    }

    #endregion
  }
}
